File: searchingAlgo/binarySearch/binarySearch.java
import java.io.*;

class BinarySearch { 
    static int search(int arr[], int value, int leftIndex, int rightIndex) {
        // base case
        if (rightIndex < leftIndex) {
            return -1;
        }

        // calculate the mid index
        int midIndex = (rightIndex - leftIndex) / 2;
        midIndex += leftIndex;

        // if value is same as value on midIndex, simply return midIndex
        if (value == arr[midIndex]) {
            return midIndex;
        } else if (value < arr[midIndex]) {
            // when value is less than middle value, we search in the left half
            return search(arr, value, leftIndex, midIndex - 1);
        } else {
            // when value is greater than middle value, we search in the right half
            return search(arr, value, midIndex + 1, rightIndex);
        }
    }

    public static void main(String args[]) {
        // sorted array to search the value in
        int arr[] = { 10, 13, 15, 22, 89, 90 };
        // value to search for
        int value = 15; 

        // find the value
        int result = search(arr, value, 0, arr.length - 1);

        // show the result
        if (result == -1) {
            System.out.println("Item not found");
        } else {
            System.out.println("Item found at index: " + result);
        }
    }
} 

File: searchingAlgo/binarySearch/binarySearch.js
function binarySearch(arr, x) {
  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    let mid = Math.round((left + right) / 2); // calculate the mid index

    // if value is same as value on midIndex, simply return midIndex
    if (arr[mid] == x) {
      return mid;
    }

    // when value is less than middle value, we search in the left half
    if (arr[mid] < x) {
      left = mid + 1;

      // when value is greater than middle value, we search in the right half
    } else {
      right = mid - 1;
    }
  }
  return -1;
}

// testCase
// sorted array to search the value in
const arr = [2, 12, 23, 34, 45, 52, 61, 75, 85, 99];
const x = 85;

console.log(binarySearch(arr, x));
